internal/telemetry/utils.go

```diff
@@ -0,0 +1,51 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2023 Datadog, Inc.
+
+// Package telemetry implements a client for sending telemetry information to
+// Datadog regarding usage of an APM library such as tracing or profiling.
+package telemetry
+
+import (
+	"testing"
+)
+
+// MockGlobalClient replaces the global telemetry client with a custom
+// implementation of TelemetryClient. It returns a function that can be deferred
+// to reset the global telemetry client to its previous value.
+func MockGlobalClient(client Client) func() {
+	globalClient.Lock()
+	defer globalClient.Unlock()
+	oldClient := GlobalClient
+	GlobalClient = client
+	return func() {
+		globalClient.Lock()
+		defer globalClient.Unlock()
+		GlobalClient = oldClient
+	}
+}
+
+// Check is a testing utility to assert that a target key value pair
```

Comment:
```suggestion
// Check is a testing utility to assert that a target key in config contains the expected value
```
Not 100% attached to this change, but my initial read of the comment confused me since the function doesn't accept a KV pair, then I realized it meant that we're asserting on the value _at_ the key in config. Feel free to change this to something else maybe?